### ISSUE TITLE
[HOLD for payment 2023-10-23] [HOLD for payment 2023-10-23] [$500] IOU - Deletion Button Remains Visible After Changing Date after  Delete the  Requested Money

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#26019]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Click on the plus sign and select "Request Money"
2. Enter the desired amount and email address for the money request
3. Go to the IOU and edit the Date
4. Click on the three dots and select "Delete"
5. Observe that the deletion button and three dots still exist, and if we delete again, an unexpected error occurs in IOU

## Expected Result:
he deletion button and three dots should disappear after deleting the requested money in any case

## Actual Result:
The deletion button remains visible after deleting the requested money and changing the date, which causes an unexpected error if we try to delete it again

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.57-5

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/fe5e4839-15b8-41fe-88de-5850a8c560bf

https://github.com/Expensify/App/assets/78819774/b73f66ae-9a57-45db-b885-518ad52a623e

**Expensify/Expensify Issue URL:**

**Issue reported by:** @Tewodros-GirmaA

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692348919083289

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019e57dec696456a6d</li>
        <li>Upwork Job ID: 1699228556598366208</li>
        <li>Last Price Increase: 2023-09-20</li>
    </ul>
</details>

**Comment by anmurali** [comment]
Will reproduce and slap the label on tomorrow

**Comment by anmurali** [comment]
Adding an external label

**Comment by anmurali** [comment]
This seems related to https://github.com/Expensify/App/issues/25997 indeed. I think the underlying issue is the same that this request thread persists and trying to delete it again throws an error. @allroundexperts what do you think? Should we maybe put this on hold for https://github.com/Expensify/App/issues/25997 and then retest?

**Comment by anmurali** [comment]
Bumped @allroundexperts on Slack [here](https://expensify.slack.com/archives/C049HHMV9SM/p1695661032873119?thread_ts=1692348919.083289&cid=C049HHMV9SM)

**Comment by anmurali** [comment]
@joelbettner this is now blocked on you. Can you review the accepted proposal so we can start to fix it?

**Comment by anmurali** [comment]
@joelbettner asking again - also bumped you on slack - https://expensify.slack.com/archives/C049HHMV9SM/p1696992985168049

**Comment by joelbettner** [comment]
@allroundexperts I agree with your assessment. @dukenv0307 I will go ahead and assign you to this issue.

**Comment by joelbettner** [comment]
@allroundexperts bump.

**Comment by dhanashree-sawant** [comment]
Hi @joelbettner , @allroundexperts it seems like selected proposal of @dukenv0307 was actually brought from #24413 which was closed without any action. If the root cause was same, would I be eligible for reporting bonus?

**Comment by joelbettner** [comment]
@anmurali I'll let you field this question: https://github.com/Expensify/App/issues/26019#issuecomment-1786485869

Sorry, @dhanashree-sawant I'm not involved with the payouts, so I don't know.

**Comment by dhanashree-sawant** [comment]
Sure that's okay, @anmurali can you check it once you are available?

**Comment by anmurali** [comment]
Payment Summary

- [ ] Reporting Bonus of $50 each to @dhanashree-sawant (as she also reported a bug that was fixed in this issue) and @tewodrosGirma)
- [ ] Contributor @dukenv0307 $500
- [ ] Contributor plus @allroundexperts $500

**Comment by dhanashree-sawant** [comment]
Thanks @anmurali for going through that issue and awarding reporting bonus to me. Actually, this as well as #24413 is reported before 30 August, can you send us 250$ offer for it? (If it is decided to provide 50$ each as we are awarding 2 reporters reporting bonus then that's okay)

**Comment by tewodrosGirmaA** [comment]
Hello, @anmurali. This matter was reported prior to August 30th. Could you please provide us with a $250 offer

**Comment by JmillsExpensify** [comment]
$750 payment approved for @allroundexperts based on [this comment](https://github.com/Expensify/App/issues/26019#issuecomment-1810300622).

**Comment by anmurali** [comment]
@dukenv0307 -  Offer [here](https://www.upwork.com/nx/wm/offer/27753893) (I will add the bonus when I pay it)

**Comment by tewodrosGirmaA** [comment]
@anmurali Thank you, I accepted it

**Comment by anmurali** [comment]
Paid.
